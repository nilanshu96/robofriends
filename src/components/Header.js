import { Component } from 'react';

class Header extends Component {

    shouldComponentUpdate(nextProps, nextState) {
        return false;
    }

    render() {
        console.log('HEADER');
        return <h1 className="f1">RoboFriends</h1>;
    }
}

export default Header;