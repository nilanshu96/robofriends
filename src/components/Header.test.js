import { render } from '@testing-library/react';
import Header from './Header';

test("compare header snapshot", () => {
    const { container, rerender } = render(<Header />);
    expect(container.innerHTML).toMatchSnapshot();

    //to check shouldComponentUpdate
    rerender(<Header />);
    expect(container.innerHTML).toMatchSnapshot();
})