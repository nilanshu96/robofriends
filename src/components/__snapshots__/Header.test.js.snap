// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compare header snapshot 1`] = `"<div><h1 class=\\"f1\\">RoboFriends</h1><button color=\\"red\\">Count: 0</button></div>"`;

exports[`compare header snapshot 2`] = `"<div><h1 class=\\"f1\\">RoboFriends</h1><button color=\\"red\\">Count: 0</button></div>"`;
